libs/features/personalization/personalization.js

```diff
@@ -245,16 +266,35 @@ function normalizeKeys(obj) {
   }, {});
 }
 
-function handleCommands(commands, manifestId, rootEl = document) {
-  commands.forEach((cmd) => {
-    if (VALID_COMMANDS.includes(cmd.action)) {
-      try {
-        const selectorEl = rootEl.querySelector(cmd.selector);
-        if (!selectorEl) return;
-        COMMANDS[cmd.action](selectorEl, cmd.target, manifestId);
-      } catch (e) {
-        console.log('Invalid selector: ', cmd.selector);
+function getSelectedElement(selector, action) {
+  try {
+    if (action.includes('pendtosection') && selector.includes('section')) {
+      const section = selector.replace('section', '');
+      if (Number.isNaN(section)) return null;
```

Comment:
Added a small update to default to section1 if they do not specify.
And agreed, we'll need to merge the logic.  I had a bit of a chicken and egg situation.  I knew your code would handle this better, but I needed something in place until I had your code since we decided to not allow CSS selectors for this action.
How ready is your PR?  I think we can merge this PR first and then I help if needed to merge my code into yours.  Or you submit your PR to merge into this branch and we get it working together before we merge this one.  I'm fine with either approach.